### DISCUSSION TITLE
pdf file contents displaying gin the browser

### DISCUSSION THREAD

**Discussion by bsweeney** [dompdf/dompdf#2639] [General]
Hello geeks,

Has any of you guys encountered an error with dompdf when previewing the pdf file contents in the browser and lots of unformatted and unnecessary text are first printed then the actual content comes down? It works very fine on localhost but have now hosted the system on a live server and getting these kind of weird behavior shown in the pics. Thanks in advance please for any help......
![Screen Shot 2021-11-13 at 02 23 14](https://user-images.githubusercontent.com/60660749/141595220-20728526-b25d-4b5a-b2a8-79458d3c4dbf.png)
![Screen Shot 2021-11-13 at 02 22 47](https://user-images.githubusercontent.com/60660749/141595282-7ccfcaf8-a987-4023-9bd3-0d5f4538b8b8.png)

_Originally posted by @Mathias-Wa-Mucani in https://github.com/dompdf/dompdf/issues/668#issuecomment-967727436_

**Comment by bsweeney** [comment]
@Mathias-Wa-Mucani that's the actual source of the PDF document. The specific reason it's being displayed in this way is difficult to say without more information. A likely cause is that the new host has output buffering enabled by default and is modifying the headers before flushing the buffer. You can try to work around the issue by disabling output buffering or clearing the buffer before using Dompdf, and halting PHP execution after streaming the document.

```
ob_end_clean();
// dompdf code goes here
exit();
```

**Reply by Mathias-Wa-Mucani** [reply]
Thanks very much for the help bro. This has worked like charm.......

**Comment by Mathias-Wa-Mucani** [comment]
Thanks please....will try it out

____________________
mathiaswamucani
